Customizing Test Network Parameters

argakiig opened the pull request:
The test network allows for customization of the following network parameters by setting the appropriate environment variable:
```
export NANO_TEST_EPOCH_1="0xffffffc000000000",
export NANO_TEST_EPOCH_2="0xfffffff800000000",
export NANO_TEST_EPOCH_2_RECV="0xfffffe0000000000",
export NANO_TEST_NODE_PORT="17075",
export NANO_TEST_RPC_PORT="17076",
export NANO_TEST_IPC_PORT="17077",
export NANO_TEST_WEBSOCKET_PORT="17078",
export NANO_TEST_MAGIC_NUMBER="RZ",
export NANO_TEST_PEER_NETWORK="peer-test.nano.org",
export NANO_TEST_GENESIS_PUB="A419090E048FAD6FED5AEE987929BC52FE431C170F7C778ACD43C8B9346AB932",
export NANO_TEST_GENESIS_BLOCK="{\n\t\"type\": \"open\",\n\t\"source\": \"A419090E048FAD6FED5AEE987929BC52FE431C170F7C778ACD43C8B9346AB932\",\n\t\"representative\": \"nano_3b1s3691b5xffzpoounrh6nurnqyaeg3g5uwgy7etiyaq6t8ogbkkithjahe\",\n\t\"account\": \"nano_3b1s3691b5xffzpoounrh6nurnqyaeg3g5uwgy7etiyaq6t8ogbkkithjahe\",\n\t\"work\": \"f6c1644a04a77566\",\n\t\"signature\": \"7830BC72126EAEE1D44CB5BAD431279751E976E46F24DE023FAA1C0FD7A75108F195DE6C8D44F26E49766CEF1FC54518EDD9292D3D278237BD96C1A84B3D1401\"\n\t}"
```
The above would be test with an altered magic, genesis, and default peer_network

Commit:
read custom from environment variables
(NANO_TEST_EPOCH_1,NANO_TEST_EPOCH_2,NANO_TEST_EPOCH_2_RECV) work thresholds
(NANO_TEST_NODE_PORT,NANO_TEST_RPC_PORT,NANO_TEST_IPC_PORT,NANO_TEST_WEBSOCKET_PORT) ports
(NANO_TEST_MAGIC_NUMBER) magic number to segregate network traffic
(NANO_TEST_PEER_NETWORK) url for default peer
(NANO_TEST_GENESIS_PUB,NANO_TEST_GENESIS_BLOCK) genesis initialization information

Commit:
include secure when linking rpc_test otherwise windows doesn't allow the test to pass

zhyatt commented on nano/lib/config.hpp:
```diff
@@ -49,6 +50,15 @@ uint8_t get_minor_node_version ();
 uint8_t get_pre_release_node_version ();
 
+std::string GetEnvOrDefault (const std::string & variable_name, const std::string & default_value);
+uint64_t GetEnvThresholdOrDefault (const std::string & variable_name, const uint64_t & default_value);
```
Should we be using camel case or lowercase with underscores instead on variables/functions?

argakiig replied:
switched back to underscore

Commit:
formatting

Commit:
various feedback and cleanup

added convenience struct for getting hex string to uint64_t via
boost::lexical_cast

Commit:
Merge branch 'develop' into custom_test

Commit:
Formatting
